src/Protocols/DfLike.hs

```diff
@@ -1,27 +1,54 @@
 {-|
-Warning: Only read this module if you're considering implementing your own
-protocol. Here be dragons.
-
 This module implements a type class 'DfLike' which serves as a generalization
 of the 'Protocols.Df.Df' protocol. Similar protocols can provide an instance
-for it and subsequently expose all the functionality implemented in this
-module. This is set up in such a way that there's little to no overhead compared
-to implementing it manually.
+for it and subsequently expose all the functionality implemented in this module.
 -}
+
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE FunctionalDependencies #-}
-{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
`UndecidableInstances` can lead to infinite loops at compile time. It's fine to use it, but you need to explain why you need it.